source/Pgghost.h

```diff
@@ -0,0 +1,139 @@
+#ifndef PGGHOST_H
+#define PGGHOST_H
+
+// #include "../../Empirical/include/emp/math/Random.hpp"
+// #include "../../Empirical/include/emp/tools/string_utils.hpp"
+// #include <set>
+// #include <iomanip> // setprecision
+// #include <sstream> // stringstream
+#include "Host.h"
+#include "SymWorld.h"
+
+
+class PggHost: public Host {
+protected:
+  double sourcepool=0;
+
+
+public:
+  PggHost(emp::Ptr<emp::Random> _random, emp::Ptr<SymWorld> _world, emp::Ptr<SymConfigBase> _config,
+  double _intval =0.0, emp::vector<emp::Ptr<Organism>> _syms = {},
+  emp::vector<emp::Ptr<Organism>> _repro_syms = {},
+  std::set<int> _set = std::set<int>(),
+  double _points = 0.0) : Host(_random, _world, _config, _intval,_syms, _repro_syms, _set, _points) { 
+    if ( _intval > 1 || _intval < -1) {
+       throw "Invalid interaction value. Must be between -1 and 1";  // Exception for invalid interaction value
```

Comment:
You don't need this line in here since the Host constructor will do it, it's just repeated code.